client/cmd.go

```diff
@@ -55,3 +58,101 @@ func ValidateCmd(cmd *cobra.Command, args []string) error {
 }
+
+// ReadPersistentCommandFlags returns a Context with fields set for "persistent"
+// flags that do not necessarily change with context. These must be checked if
+// the caller explicitly changed the values.
+func ReadPersistentCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Context, error) {
+	if flagSet.Changed(flags.FlagTrustNode) {
+		trustNode, err := flagSet.GetBool(flags.FlagTrustNode)
+		if err != nil {
+			return clientCtx, err
+		}
+
+		clientCtx = clientCtx.WithTrustNode(trustNode)
+	}
+
+	if flagSet.Changed(flags.FlagKeyringBackend) {
+		keyringBackend, err := flagSet.GetString(flags.FlagKeyringBackend)
+		if err != nil {
+			return clientCtx, err
+		}
+
+		kr, err := newKeyringFromFlags(clientCtx, keyringBackend)
+		if err != nil {
+			return clientCtx, err
+		}
+
+		clientCtx = clientCtx.WithKeyring(kr)
+	}
+
+	if flagSet.Changed(flags.FlagNode) {
+		rpcURI, err := flagSet.GetString(flags.FlagNode)
+		if err != nil {
+			return clientCtx, err
+		}
+
+		clientCtx = clientCtx.WithNodeURI(rpcURI)
+	}
+
+	return clientCtx, nil
+}
+
+// ReadQueryCommandFlags returns an updated Context with fields set based on flags
+// defined in GetCommands. An error is returned if any flag query fails.
+//
+// Certain flags are naturally command and context dependent, so for these flags
+// we do not check if they've been explicitly set by the caller. Other flags can
+// be considered "persistent" (e.g. KeyBase or Client) and these should be checked
+// if the caller explicitly set those.
+func ReadQueryCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Context, error) {
+	height, _ := flagSet.GetInt64(flags.FlagHeight)
+	clientCtx = clientCtx.WithHeight(height)
+
+	useLedger, _ := flagSet.GetBool(flags.FlagUseLedger)
+	clientCtx = clientCtx.WithUseLedger(useLedger)
+
+	return ReadPersistentCommandFlags(clientCtx, flagSet)
+}
+
+// ReadTxCommandFlags returns an updated Context with fields set based on flags
+// defined in PostCommands. An error is returned if any flag query fails.
+//
+// Certain flags are naturally command and context dependent, so for these flags
+// we do not check if they've been explicitly set by the caller. Other flags can
+// be considered "persistent" (e.g. KeyBase or Client) and these should be checked
+// if the caller explicitly set those.
+func ReadTxCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Context, error) {
+	clientCtx, err := ReadPersistentCommandFlags(clientCtx, flagSet)
+	if err != nil {
+		return clientCtx, err
+	}
+
+	genOnly, _ := flagSet.GetBool(flags.FlagGenerateOnly)
+	clientCtx = clientCtx.WithGenerateOnly(genOnly)
+
+	dryRun, _ := flagSet.GetBool(flags.FlagDryRun)
+	clientCtx = clientCtx.WithSimulation(dryRun)
+
+	offline, _ := flagSet.GetBool(flags.FlagOffline)
+	clientCtx = clientCtx.WithOffline(offline)
+
+	useLedger, _ := flagSet.GetBool(flags.FlagUseLedger)
+	clientCtx = clientCtx.WithUseLedger(useLedger)
+
+	bMode, _ := flagSet.GetString(flags.FlagBroadcastMode)
```

Comment:
Can't broadcast mode be set via a config file?

Reply:
Not via the app.toml. I would like to deprecate the current `config.toml` that the CLI currently supports. It will clobber with the existing configuration.

Reply:
Not to mention it also relies on a global viper.

Reply:
What do you propose as a replacement for the current config?

Reply:
Removed entirely. See https://github.com/cosmos/cosmos-sdk/pull/6525#issuecomment-651973322.